// ngx-web-storage/__ivy_ngcc__/dist/web-storage.module.js
"use strict";
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@angular/common');
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var web_storage_config_1 = require("./web-storage.config");
var web_storage_service_1 = require("./core/web-storage.service");
var WebStorageModule = /** @class */ (function () {
    function WebStorageModule(parentModule) {
        if (parentModule) {
            throw new Error('StorageModule is already loaded. Import it in the root module only');
        }
    }
    WebStorageModule_1 = WebStorageModule;
    WebStorageModule.forRoot = function (config) {
        web_storage_config_1.ConfigManager.set(config);
        return {
            ngModule: WebStorageModule_1,
        };
    };
    var WebStorageModule_1;
    WebStorageModule = WebStorageModule_1 = __decorate([ __param(0, core_1.Optional()), __param(0, core_1.SkipSelf()),
        __metadata("design:paramtypes", [WebStorageModule])
    ], WebStorageModule);
WebStorageModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WebStorageModule });
WebStorageModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WebStorageModule_Factory(t) { return new (t || WebStorageModule)(ɵngcc0.ɵɵinject(WebStorageModule, 12)); }, providers: [
        web_storage_service_1.WebStorageService,
    ], imports: [[
            common_1.CommonModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WebStorageModule, { imports: [ɵngcc1.CommonModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WebStorageModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [
                    common_1.CommonModule,
                ],
                providers: [
                    web_storage_service_1.WebStorageService,
                ]
            }]
    }], function () { return [{ type: WebStorageModule, decorators: [{
                type: core_1.Optional
            }, {
                type: core_1.SkipSelf
            }] }]; }, null); })();
    return WebStorageModule;
}());
exports.WebStorageModule = WebStorageModule;
